File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03253/p03253_num4_parsed.py
def count_sequences(N, M):
    MOD = 10**9 + 7
    dp = [[0] * (M+1) for _ in range(N+1)]
    dp[0][1] = 1
    for i in range(1, N+1):
        for j in range(1, M+1):
            dp[i][j] = (dp[i][j-1] + dp[i-1][j//2]) % MOD
    return dp[N][M]

N, M = map(int, input().split())
print(count_sequences(N, M))

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03253/p03253_num4_parsed.cpp
#include <iostream>
#include <vector>

const int MOD = 1000000007;

int countSequences(int N, int M) {
    std::vector<std::vector<int>> dp(N + 1, std::vector<int>(M + 1, 0));
    dp[0][1] = 1;
    
    for (int i = 1; i <= N; i++) {
        for (int j = 1; j <= M; j++) {
            for (int k = 1; k <= j; k++) {
                if (j % k == 0) {
                    dp[i][j] = (dp[i][j] + dp[i - 1][j / k]) % MOD;
                }
            }
        }
    }
    
    int result = 0;
    for (int i = 1; i <= M; i++) {
        result = (result + dp[N][i]) % MOD;
    }
    
    return result;
}

int main() {
    int N, M;
    std::cin >> N >> M;
    
    int result = countSequences(N, M);
    std::cout << result << std::endl;

    return 0;
}